index.js

```diff
@@ -237,6 +241,30 @@ client.on("messageCreate", async (message) => {
     }
 });
 
+// Event listener for voice calls:
+client.on("voiceStateUpdate", (oldState, newState) => {
+    const userId = newState.member.id;
+
+    // User joins voice channel:
+    if (!oldState.channel && newState.channel) {
+        callRecords[userId] = {
+            startTime: Date.now(),
+            timeout: setTimeout(() => {
+                const voiceChannel = newState.channel;
+                const targetChannel = client.channels.cache.get(GENERAL_CHANNEL_ID);
+                if (voiceChannel instanceof VoiceChannel) {
+                    targetChannel.send(`Remember to take notes!`);
```

Comment:
Hmm, is there a way to refactor this so that rather than doing it by user ID, we do it by call ID? For example, when a call starts, we start a 5 minute timer. Then as users join, you add it to a list of users for that call ID. If they leave, remove them from that list. Once we hit the 5-minute mark, then it pings ALL of the users in one ping.

Right now, the current implementation means every time a new user joins and they've been in the call for 5 mins, they'll get pinged. So if there are 100 users on the call, this will be 100 separate pings, one for each user, and potentially even more if a user had to leave for a few seconds or got disconnected and had to rejoin

Reply:
Good point. Will refactor in PR